### PR TITLE
Add libSQL support

### PR DESCRIPTION
**Pull request by psarna**
This pull request adds `bundled-libsql` and `bundled-libsql-wasm` features, which allow using yet another bundled library - libSQL. The changes were inspired by the `bundled-sqlcipher` feature, which already allows using a fork of SQLite instead of the vanilla one.

The `-wasm` feature also enables WebAssembly user-defined functions support. It's moved to a separate feature, because the Wasm engine needs to be linked separately via the https://crates.io/crates/libsql-wasmtime-bindings crate. Example: https://github.com/libsql/sqld/blob/95f00661b28c9eb88e2cb0c7a686027cd7a77126/sqld-libsql-bindings/src/lib.rs#L8-L10 , and it's an additional dependency that not everybody needs.

### CODE REVIEW

**Comment by psarna** [comment]
(One obvious fix on libSQL end is to fix including ext/udf/wasm_bindings.h into the sqlite3.h amalgamation file, I'll do that soon)

**Comment by psarna** [comment]
With [libSQL 0.2.0 released](https://github.com/libsql/libsql/releases/tag/libsql-0.2.0), I do not consider this PR a draft anymore, marking as ready for review!

**Comment by psarna** [comment]
hi again, let me know if I can help in proceeding with this PR!

**Comment by thomcc** [comment]
Hey, let me get back to you shortly. (Since we chatted last week I've become somewhat busier than anticipated with both my day job and obligations for the Rust project, but should have time to respond more soon)

File: libsqlite3-sys/libsql/bindgen_bundled_version.rs [hunk 1]
```diff
@@ -0,0 +1,3388 @@
+/* automatically generated by rust-bindgen 0.64.0 */
+
+pub const __GNUC_VA_LIST: i32 = 1;
```
Comment (thomcc) [inline libsqlite3-sys/libsql/bindgen_bundled_version.rs:3]: You should tweak the bindgen usage so that stuff like this doesn't show up.

**Review by thomcc — CHANGES_REQUESTED** [review]
Basically looks fine.

My main concern is around exposing experimental functionality here, since I'd like to get libsqlite3-sys to 1.0 in the forseeable future (I haven't had time to work on it lately, admittedly...), and that means the libsql-specific APIs concern me.

But, do you mind putting the libsql-specific APIs behind a `libsql-experimental` flag in both libsqlite3-sys and rusqlite?

File: libsqlite3-sys/libsql/bindgen_bundled_version.rs (hunk 1 above)
Comment (psarna) [reply libsqlite3-sys/libsql/bindgen_bundled_version.rs:3]: Will do, do you have any hints how? I actually produced all the pregenerated bindings by building libsqlite3-sys with `buildtime_bindgen`, and then taking the bindings straight from the target/release/build/ directory (:

**Comment by psarna** [comment]
> But, do you mind putting the libsql-specific APIs behind a libsql-experimental flag in both libsqlite3-sys and rusqlite?
No problem, but let me just clarify first. I put all the new code under `libsql` and `libsql-wasm` features already. So is it alright if I just rename the feature to `libsql-experimental`, to make it obvious the APIs are new and not yet as battle-tested as vanilla SQLite? Or do you have some more fine-grained change in mind, like using both `libsql` and `libsql-experimental` feature flags in the code?

**Comment by psarna** [comment]
I pushed an additional patch on top, one that adds `-experimental` suffix to the features. Let me know if it looks alright

File: libsqlite3-sys/libsql/bindgen_bundled_version.rs (hunk 1 above)
Comment (thomcc) [reply libsqlite3-sys/libsql/bindgen_bundled_version.rs:3]: Well, it's not clear to me why this isn't triggering: https://github.com/rusqlite/rusqlite/blob/master/libsqlite3-sys/build.rs#L592

**Comment by thomcc** [comment]
> No problem, but let me just clarify first. I put all the new code under libsql and libsql-wasm features already. So is it alright if I just rename the feature to libsql-experimental

Yes, that's fine.

File: libsqlite3-sys/libsql/bindgen_bundled_version.rs (hunk 1 above)
Comment (thomcc) [reply libsqlite3-sys/libsql/bindgen_bundled_version.rs:3]: See the upgrade.sh scripts for how we usually do this.

Comment (psarna) [reply libsqlite3-sys/libsql/bindgen_bundled_version.rs]: Just reran bindings generation with `LIBSQLITE3_SYS_BUNDLING` env variable set, the diff looks like it worked exactly as it should: https://github.com/rusqlite/rusqlite/pull/1283/commits/3a7d0b6988b7bcc1550fbd447fde62d07799051d

**Comment by sunny-g** [comment]
@psarna is this still being worked on?

**Pull request closed without merging**